src/app/(drift)/(auth)/signin/page.tsx

```diff
@@ -2,14 +2,18 @@ import { getMetadata } from "src/app/lib/metadata"
 import config from "@lib/config"
 import Auth from "../components"
 
-function isGithubEnabled() {
-	return config.github_client_id.length && config.github_client_secret.length
-		? true
-		: false
+function isGithubEnabled(): boolean {
+	return !!(config.github_client_id && config.github_client_secret)
+}
+
+function isKeycloakEnabled(): boolean {
+	return !!(config.keycloak_client_id
+		&& config.keycloak_client_secret
+		&& config.keycloak_issuer)
 }
 
 export default function SignInPage() {
-	return <Auth page="signin" isGithubEnabled={isGithubEnabled()} />
+	return <Auth page="signin" isGithubEnabled={isGithubEnabled()} isKeycloakEnabled={isKeycloakEnabled()} />
```

Comment:
this is going to get messy fast. I'll revisit this after this is merged though

Reply:
Yes, I will try to make it better ;).